iceoryx_posh/test/integrationtests/test_posh_experimental_node.cpp

```diff
@@ -388,4 +498,46 @@ TEST(Node_test, NodeAndEndpointsAreContinuouslyRecreated)
     }
 }
 
+TEST(Node_test, MultipleNodeAndEndpointsAreRegisteredWithSeparateRouDiRunningInParallel)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1e527815-28d1-4a99-a9a3-cc4084018cf3");
+
+    NodeName_t node_name{"hypnotoad"};
+    ServiceDescription service_description{"all", "glory", "hypnotoad"};
+
+    constexpr uint16_t domain_id_a{13};
+    constexpr uint16_t domain_id_b{42};
+
+    RouDiEnv roudi_a{DomainId{domain_id_a}};
```

Comment:
There is a test in `test_roudi_env.cpp` but it panics. The use case is different though since one usually does not start RouDi in the same process like the applications but in a separate process.

If one wants to have a more graceful error handling for RouDi, one needs to adapt `IceOryxRouDiMemoryManager` to not panic in the ctor.

It is out of scope for this PR though.